mantidproject/mantid
# Added GIL release to findRuns python calls

**Description of work.**

This adds a class to release the python GIL and then uses it to release the GIL before executing FileFinder.findRuns calls.

**To test:**

Run the following script after setting the instrument to LOQ and enabling the ISIS data archive.
```
from mantid.api import FileFinder

FileFinder.findRuns(''12345678'')
```
and check that the GUI does not freeze whilst it is running.


Fixes #23266 . 


*This does not require release notes* because this is not a user facing change.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
